doc/example_pipelines.md

```diff
@@ -17,7 +17,7 @@ executes the example with `--help` to check that it runs.
 You can test it using a clone of the OCurrent repository itself:
 
 ```
-$ dune exec -- examples/docker_build_local.exe .
+$ dune exec -- doc/examples/docker_build_local.exe .
```

Comment:
At the risk of bike-shedding a very small change, it's a shame to refer to the directory name inside that directory. The executables are public, so we could have:
```suggestion
$ dune exec -- doc/examples/docker_build_local.exe .
```
```suggestion
$ dune exec -- docker_build_local .
```
We could go further and allow the root directory to be inferred by git:
```diff
diff --git a/doc/examples/docker_build_local.ml b/doc/examples/docker_build_local.ml
index f1f8b34..47ccf4d 100644
--- a/doc/examples/docker_build_local.ml
+++ b/doc/examples/docker_build_local.ml
@@ -20,15 +20,16 @@ let pipeline ~repo () =
 [@@@part "end-pipeline"]

 let main config mode repo =
-  let repo = Git.Local.v (Fpath.v repo) in
-  let engine = Current.Engine.create ~config (pipeline ~repo) in
-  let site = Current_web.Site.(v ~has_role:allow_all) ~name:program_name (Current_web.routes engine) in
-  Lwt_main.run begin
-    Lwt.choose [
-      Current.Engine.thread engine;
-      Current_web.run ~mode site;
-    ]
-  end
+  let main =
+    let open Lwt.Infix in
+    let cmd = [| "git"; "-C"; repo; "rev-parse"; "--show-toplevel" |] in
+    Lwt_process.pread ("", cmd) >>= fun repo ->
+      let repo = Git.Local.v (Fpath.v (String.trim repo)) in
+      let engine = Current.Engine.create ~config (pipeline ~repo) in
+      let site = Current_web.Site.(v ~has_role:allow_all) ~name:program_name (Current_web.routes engine) in
+      Lwt.choose [Current.Engine.thread engine; Current_web.run ~mode site]
+  in
+  Lwt_main.run main
```

Reply:
I agree with your first suggestion! It will be lighter to keep just `docker_build_local`.
However, IMHO, it might be confusing (as it's the first example) to add some extra git command to it...

Reply:
Perhaps if the git trick is separated from main, to keep the example a little cleaner. That's then slightly less invasive (ignoring space changes):
```diff
diff --git a/doc/examples/docker_build_local.ml b/doc/examples/docker_build_local.ml
index f1f8b34..54a7e08 100644
--- a/doc/examples/docker_build_local.ml
+++ b/doc/examples/docker_build_local.ml
@@ -19,14 +19,21 @@ let pipeline ~repo () =

 [@@@part "end-pipeline"]

+open Lwt.Infix
+
+let find_git_root dir =
+  let cmd = [| "git"; "-C"; dir; "rev-parse"; "--show-toplevel" |] in
+  Lwt_process.pread ("", cmd) >|= String.trim
+
 let main config mode repo =
+  Lwt_main.run begin
+    find_git_root repo >>= fun repo ->
       let repo = Git.Local.v (Fpath.v repo) in
       let engine = Current.Engine.create ~config (pipeline ~repo) in
       let site = Current_web.Site.(v ~has_role:allow_all) ~name:program_name (Current_web.routes engine) in
-  Lwt_main.run begin
       Lwt.choose [
         Current.Engine.thread engine;
         Current_web.run ~mode site;
       ]
   end
```

Reply:
I agree with your proposition: I push the changes.